### ISSUE TITLE
Help me !!! loss = nan help

### ISSUE THREAD

**Issue by ninenylele** [keras-team/keras#18532]
model = Sequential()
model.add(Conv2D(512, (5, 5), input_shape=(64, 64, 1), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Flatten())
model.add(Dense(250, activation='relu'))
model.add(Dense(num_classes, activation='softmax'))
#Choose an optimizer and compile the model.
model.compile(optimizer = Adam(learning_rate = 0.001#i use 0.01 loss = nan same ), loss = 'categorical_crossentropy', metrics = ['accuracy'])
#And print the summary of the model.
print("Model Summary : ")
print(model.summary())
model1 = model.fit(x_train, y_train, batch_size=64, epochs=50)
Epoch 1/5
176/176 [==============================] - 9s 47ms/step - loss: nan - accuracy: 0.4943

**Comment by kernel-loophole** [comment]
can you share colab notebook for this?

**Comment by Adesoji1** [comment]
@ninenylele   The `nan` (Not a Number) in your training loss suggests that there's an issue with the model training. This is often due to numerical instability which can arise from various reasons. Here are some steps you can take to troubleshoot:

1. **Learning Rate**: You mentioned that you've tried `0.01` and now `0.001`. Sometimes, even smaller learning rates might be needed to ensure stability, especially when using deeper models. Try reducing the learning rate further, like `0.0001` or `0.00001`.

2. **Weight Initialization**: Poor weight initialization can lead to vanishing or exploding gradients. By default, Keras uses `glorot_uniform` (also known as Xavier initialization) for its layers, which usually works fine for relatively shallow networks. But sometimes, other initializations like `he_normal` can work better, especially for networks with ReLU activations. 

   ```python
   model.add(Conv2D(512, (5, 5), input_shape=(64, 64, 1), activation='relu', kernel_initializer='he_normal'))
   ```

3. **Batch Normalization**: Adding batch normalization can stabilize the activations and gradients, making the network more robust.

   ```python
   from keras.layers import BatchNormalization
   model.add(BatchNormalization())
   ```

   Insert this after your Conv2D or Dense layers and before activation.

4. **Check Input Data**: Ensure that your training data `x_train` is normalized, typically to have values between [0, 1] (if your images are in the format of uint8) or [-1, 1]. If your data has very large or very small values, it can cause instability.

   ```python
   x_train = x_train.astype('float32') / 255.0
   ```

5. **Reduce Model Complexity**: You start with a very large number of filters (512) in the first layer for a `(64,64,1)` input. This might be too much capacity for the model initially. Try reducing the number of filters to see if it stabilizes the training.

6. **Check Target Data**: Ensure that your `y_train` labels are correctly one-hot encoded and don't contain any NaN values.

7. **Gradient Clipping**: In some cases, especially with RNNs, gradients can become very large and "explode", causing NaN values. Gradient clipping can help, but it's a less common solution for CNNs.

These are my thoughts and i stand to be corrected. By iteratively making these adjustments and monitoring the training loss, you should be able to identify and resolve the cause of the NaN loss.

**Comment by SuryanarayanaY** [comment]
Hi @ninenylele,

Could you please submit reproducible code snippet ? Thanks!
